Library/Homebrew/version.rb

```diff
@@ -179,6 +179,19 @@ def self.detect(url, specs)
     end
   end
 
+  # TODO add tests if method is going to be left
+  def self.parse_version(val)
```

Comment:
`parse_version` is called only in those places where `Version.new` used to be called, `parse` and `parse_version` are for different purposes, I think. `parse` is more helpful for parsing url and different `Pathname`'s, but `parse_version` is only used to decide whether we need a `Version` or `HeadVersion`. But I'll merge if you still think it's better.